File: js/iterators_generators/doklad/python/generator.py
import random

def gen_func():
  while True:
    yield random.random()

generator = gen_func()

print('generator function', gen_func) # генераторная функция
print('generator', generator)     # генератор

print('value: ', generator.__next__())
print('value: ', generator.__next__())
print('value: ', generator.__next__())

# OUTPUT:
# generator function <function gen_func at 0x7fe71d848c80>
# generator <generator object gen_func at 0x7fe71d7732d0>
# value:  0.5982845385357144
# value:  0.3308434359315522
# value:  0.3165169253705432

File: js/iterators_generators/doklad/javascript/generator.js
function* genFunc() {
	while(true) {
		yield Math.random(1);
	}
}

const generator = genFunc();

console.log('value:', generator.next());
console.log('value:', generator.next());
console.log('value:', generator.next());
console.log('value:', generator.next());
console.log('value:', generator.next());
console.log('value:', generator.next());

console.log('generator func', genFunc);
console.log('generator', generator);


// OUTPUT:

// value: { value: 0.921832164052985, done: false }
// value: { value: 0.418734620386483, done: false }
// value: { value: 0.3309010343818355, done: false }
// value: { value: 0.1657775354136879, done: false }
// value: { value: 0.890489361265606, done: false }
// value: { value: 0.554218046878145, done: false }

// generator func function* genFunc() {
// 	while(true) {
// 		yield Math.random(1);
// 	}
// }
// generator Object [Generator] {}